app/repository/document.py

```diff
@@ -125,7 +125,7 @@ def _document_tuple_from_dto(db: Session, dto: DocumentCreateDTO) -> CreateObjec
     phys_doc = PhysicalDocument(
         id=None,
         title=dto.title,
-        source_url=dto.source_url,
+        source_url=str(dto.source_url),
```

Comment:
Here you are creating a new string based on the object (which itself is a string) - if you use `cast()` this is just generating a new memoryview. Overall this should have no effect, but its worth noting what you've written here could allocate more memory un-necessarily.

Reply:
Oooo, this is really good to know!

Reply:
I am confusing with cast() from typing.
I did the changes you mentioned (str() -> cast(str, ...)), but these are not working
I have read about better about cast, and basically cast is a function to make the linter trust in the type when these are not match, but do not change the type.
Therefore, the variables are being still URL from pydantic, and sqlalchemy is still not understanding them...
 
`sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) can't adapt type 'pydantic_core._pydantic_core.Url'`

It's important to note that cast() does not perform any runtime conversion operation